### ISSUE TITLE
Not found order details is not handled [info provided]

### ISSUE THREAD

**Issue by mateuszo** [SAP/spartacus#11433]
Spartacus doesn't handle not-found orders. When opening an order details page with an order id that doesn't exist blank order details page appears. It is not possible to reproduce this when browsing from the order history list with consistent data. However, there are other scenarios when a user could face this issue: 
* Changing an account and trying to open previously saved order details URL.
* Inconsistency/missing data/temporary unavailability of the API.

- Library version: 3.0

**To Reproduce**
1. Go to 'my-account/order/wrong-order-id'
2. Observe empty order details page

**Expected behavior**
Some kind of handling should be implemented. It could be a 404 page or special order not found page. 

**Screenshots**
![image](https://user-images.githubusercontent.com/2676966/110471581-940d6b80-80dc-11eb-901f-fa733d71fa90.png)

**Additional context**
In the project we're working on we've implemented a workaround that redirects a user to the 404 page when the order details API returns an error. This causes another issue that the user cannot press the back button to get to the order list because it redirects to 404 endlessly. A dirty solution would be to manipulate the browser history. Is it possible to change the current page in Spartacus without routing?

**Comment by giancorderoortiz** [comment]
Attempt to reproduce.

**Comment by dydome** [comment]
investigating

**Comment by giancorderoortiz** [comment]
Issue continue to happens with 4.3.x. 
Will it be fixed for 5.0 ? 
@KateChuen, @Xymmer fyi

**Comment by giancorderoortiz** [comment]
![Screen Shot 2022-01-31 at 2 40 37 PM](https://user-images.githubusercontent.com/46171897/151861217-8940d999-7659-4ac9-975a-888a63bc225a.png)

**Comment by Xymmer** [comment]
can we backport after we fix in 5.0?

**Comment by dydome** [comment]
After a few days of analysis and checking a few different ideas and solutions with other devs and architects. We decided to leave this task for a while and wait for another Order lib refactor. Currently getting order details is handled by asynchronous order load method based on ngrx and synchronous get loaded details method. There are a few possibilities to fix this problem:

- protect all order details / order replenishment / order approval templates/views from showing missing data
- replace getOrderDetails with getOrderDetailsState but this approach requires also some template changes and many places to change
- create guards for details components, for this approach we also need to get details state to see when order details data is available as in is no synchronous

All of them are complex changes but just fixing one problem and not refactoring the unlucky order ngrx strategy. We decided that the best way is to change the Order lib approach from ngrx to commands but this is a huge change and will be performing with other Order Lib changes.
